Build failure: subversionClient on aarch64-darwin

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `subversionClient`

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
error: builder for '/nix/store/rp0i3wrcwj8j9f69nyd728mv9a074cg0-subversion-client-1.14.5.drv' failed with exit code 2;
       last 25 log lines:
       >       |                 ~~~~~~~~~~~~~~~~~~~~~~~~                                              ^
       > subversion/bindings/swig/python/svn_client.c:1258:1: note: 'SWIG_Python_AppendOutput' declared here
       >  1258 | SWIG_Python_AppendOutput(PyObject* result, PyObject* obj) {
       >       | ^                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > subversion/bindings/swig/python/svn_client.c:9168:78: error: too many arguments to function call, expected 2, have 3
       >  9168 |     resultobj = SWIG_Python_AppendOutput(resultobj, SWIG_From_long((*arg1)), 0);
       >       |                 ~~~~~~~~~~~~~~~~~~~~~~~~                                     ^
       > subversion/bindings/swig/python/svn_client.c:1258:1: note: 'SWIG_Python_AppendOutput' declared here
       >  1258 | SWIG_Python_AppendOutput(PyObject* result, PyObject* obj) {
       >       | ^                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > subversion/bindings/swig/python/svn_client.c:9171:116: error: too many arguments to function call, expected 2, have 3
       >  9171 |     resultobj = SWIG_Python_AppendOutput(resultobj, SWIG_NewPointerObj((void*)(arg1), SWIGTYPE_p_long, new_flags), 0);
       >       |                 ~~~~~~~~~~~~~~~~~~~~~~~~                                                                           ^
       > subversion/bindings/swig/python/svn_client.c:1258:1: note: 'SWIG_Python_AppendOutput' declared here
       >  1258 | SWIG_Python_AppendOutput(PyObject* result, PyObject* obj) {
       >       | ^                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > subversion/bindings/swig/python/svn_client.c:9315:78: error: too many arguments to function call, expected 2, have 3
       >  9315 |     resultobj = SWIG_Python_AppendOutput(resultobj, SWIG_From_long((*arg1)), 0);
       >       |                 ~~~~~~~~~~~~~~~~~~~~~~~~                                     ^
       > subversion/bindings/swig/python/svn_client.c:1258:1: note: 'SWIG_Python_AppendOutput' declared here
       >  1258 | SWIG_Python_AppendOutput(PyObject* result, PyObject* obj) {
       >       | ^                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > fatal error: too many errors emitted, stopping now [-ferror-limit=]
       > 5 warnings and 20 errors generated.
       > make: *** [build-outputs.mk:2607: subversion/bindings/swig/python/svn_client.lo] Fehler 1
       For full logs, run 'nix log /nix/store/rp0i3wrcwj8j9f69nyd728mv9a074cg0-subversion-client-1.14.5.drv'.
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->
Build failure on Hydra: https://hydra.nixos.org/build/282891753

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.2.0, macOS 15.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixpkgs"`
 - channels(rocko): `"nixpkgs, home-manager"`
 - nixpkgs: `/nix/store/6zgbbqlr7nnfxpzkyj7fsl4fpg89jbw0-source`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@lovek323 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/version-management/subversion/default.nix
+++ b/pkgs/applications/version-management/subversion/default.nix
@@ -94,6 +94,12 @@ let
 
         patches = [ ./apr-1.patch ] ++ extraPatches;
 
+        # remove vendored swig-3 files as these will shadow the swig provided
+        # ones and result in compile errors
+        postPatch = ''
+          rm subversion/bindings/swig/proxy/{perlrun.swg,pyrun.swg,python.swg,rubydef.swg,rubyhead.swg,rubytracking.swg,runtime.swg,swigrun.swg}
+        '';
+
         # We are hitting the following issue even with APR 1.6.x
         # -> https://issues.apache.org/jira/browse/SVN-4813
         # "-P" CPPFLAG is needed to build Python bindings and subversionClient

